migration-3.0.md

```diff
@@ -0,0 +1,156 @@
+# Migrating from PER-CS v1.0 (PSR-12) to PER-CS v2.0 ###
+
+## Summary
+
+PER-CS is the next evolution of the PSR set of Coding Standards from the
+PHP-FIG (Framework Interoperability Group). It extends the Coding Standards
+laid out in PSR-12 to the newest functionality added to PHP such as the match
+keyword, enums, attributes, and more.
+
+This document describes the changes and additions on a section by section
+basis between PER-CS v3.0 and PER-CS v2.0.
+
+It is derived in part from [a GitHub-generated diff](https://github.com/php-fig/per-coding-style/compare/2.0.0...3.0.0#files_bucket)
+and focuses on the changes on a section-by-section basis as its focus is to be more readable.
+
+This document intends to provide a summary of these changes that can
+then be used to drive action lists for toolset producers to support PER-CS v3.0.
+
+This document is non-normative.  The published [3.0 PER-CS](https://www.php-fig.org/per/coding-style/) specification 
+is the canonical source for the PER-CS formatting expectations.
+
+## [Section 2.5 - Keywords and Types](https://www.php-fig.org/per/coding-style/#25-keywords-and-types)
+
+Formatting conventions are now provided for compound types (those that include union or intersection type declarations).
+
+* The `|` and `&` symbols and parentheses MUST NOT have leading or trailing spaces.
+* If a type declaration is long enough to split to multiple lines, each ANDed block must be on one line, and each ORed block on its own line.
```

Comment:
```suggestion
* If a type declaration is long enough to split to multiple lines, each block must be on one line, and each block on its own line.
```

Reply:
I'm not sure this is clear for what it implies.  There's 2 different kinds of blocks, so it's not clear what this version is saying.

Reply:
Could benefit from an illustrative snippet.

Reply:
There may be a misunderstanding on my part with this suggestion.
<br>
The original texts says: `[...] each ANDed bock must [...]` and `[...] and each ORed block on [...]`.

I thought it was a typo, and removing these two words from the text (`ANDed` & `ORed`) makes sense.
If it is the case that these words are specific blocks or something, all good 👍